libfreerdp/crypto/crypto.c

```diff
@@ -289,6 +289,30 @@ BYTE* crypto_cert_hash(X509* xcert, const char* hash, UINT32* length)
 	return fp;
 }
 
+char* cryto_cert_hash_use_signature_alg(X509* xcert)
+{
+	WINPR_ASSERT(xcert);
+
+	const ASN1_OBJECT* sigAlgObj;
+	const X509_ALGOR* tsig_alg = X509_get0_tbs_sigalg(xcert);
+	char* hashName;
+
+	hashName = calloc(50, sizeof(char));
+	if (hashName == NULL)
+		return NULL;
+
+	X509_ALGOR_get0(&sigAlgObj, NULL, NULL, tsig_alg);
+	OBJ_obj2txt(hashName, 50, sigAlgObj, 0);
+
+	/* OBJ_obj2txt returns hashWithEncryption, ex. sha256WithRSAEncrpytion;
+	 * we need only the part before With */
+	char* w = strstr(hashName, "With");
```

Comment:
this check is not sufficient, it only covers `RSA` certificates.
Might be better to use `X509_get_signature_nid` and do a big switch for all (currently) supported algorighm combinations (maybe there is something simpler?)